libraries/AP_Vehicle/AP_Vehicle.cpp

```diff
@@ -1111,6 +1111,50 @@ void AP_Vehicle::check_motor_noise()
 #endif
 }
 
+#if HAL_WITH_ESC_TELEM && (APM_BUILD_COPTER_OR_HELI || APM_BUILD_TYPE(APM_BUILD_ArduPlane))
+bool AP_Vehicle::motors_takeoff_check(float rpm_min, float rpm_max)
```

Comment:
Could this whole function live in ESC telem lib?

Reply:
This was one of the decisions from https://github.com/ArduPilot/ardupilot/pull/27696. It was originally put within AP_Motors, but because it depends on two subsystems, it was decided that it should go higher up to AP_Vehicle.